packages/deployments/contracts/contracts/messaging/libraries/Merkle.sol

```diff
@@ -151,91 +843,170 @@ library MerkleLib {
+      _current := keccak256(0, 0x40)
+
+      f := shl(5, and(_index, shl(16, 1)))
+      mstore(f, _current)
+      mstore(sub(0x20, f), mload(add(BRANCH_DATA_OFFSET, shl(5, 16))))
+      _current := keccak256(0, 0x40)
+
+      f := shl(5, and(_index, shl(17, 1)))
+      mstore(f, _current)
+      mstore(sub(0x20, f), mload(add(BRANCH_DATA_OFFSET, shl(5, 17))))
+      _current := keccak256(0, 0x40)
+
+      f := shl(5, and(_index, shl(18, 1)))
+      mstore(f, _current)
+      mstore(sub(0x20, f), mload(add(BRANCH_DATA_OFFSET, shl(5, 18))))
+      _current := keccak256(0, 0x40)
+
+      f := shl(5, and(_index, shl(19, 1)))
+      mstore(f, _current)
+      mstore(sub(0x20, f), mload(add(BRANCH_DATA_OFFSET, shl(5, 19))))
+      _current := keccak256(0, 0x40)
+
+      f := shl(5, and(_index, shl(20, 1)))
+      mstore(f, _current)
+      mstore(sub(0x20, f), mload(add(BRANCH_DATA_OFFSET, shl(5, 20))))
+      _current := keccak256(0, 0x40)
+
+      f := shl(5, and(_index, shl(21, 1)))
+      mstore(f, _current)
+      mstore(sub(0x20, f), mload(add(BRANCH_DATA_OFFSET, shl(5, 21))))
+      _current := keccak256(0, 0x40)
+
+      f := shl(5, and(_index, shl(22, 1)))
+      mstore(f, _current)
+      mstore(sub(0x20, f), mload(add(BRANCH_DATA_OFFSET, shl(5, 22))))
+      _current := keccak256(0, 0x40)
+
+      f := shl(5, and(_index, shl(23, 1)))
+      mstore(f, _current)
+      mstore(sub(0x20, f), mload(add(BRANCH_DATA_OFFSET, shl(5, 23))))
+      _current := keccak256(0, 0x40)
+
+      f := shl(5, and(_index, shl(24, 1)))
+      mstore(f, _current)
+      mstore(sub(0x20, f), mload(add(BRANCH_DATA_OFFSET, shl(5, 24))))
+      _current := keccak256(0, 0x40)
+
+      f := shl(5, and(_index, shl(25, 1)))
+      mstore(f, _current)
+      mstore(sub(0x20, f), mload(add(BRANCH_DATA_OFFSET, shl(5, 25))))
+      _current := keccak256(0, 0x40)
+
+      f := shl(5, and(_index, shl(26, 1)))
+      mstore(f, _current)
+      mstore(sub(0x20, f), mload(add(BRANCH_DATA_OFFSET, shl(5, 26))))
+      _current := keccak256(0, 0x40)
+
+      f := shl(5, and(_index, shl(27, 1)))
+      mstore(f, _current)
+      mstore(sub(0x20, f), mload(add(BRANCH_DATA_OFFSET, shl(5, 27))))
+      _current := keccak256(0, 0x40)
+
+      f := shl(5, and(_index, shl(28, 1)))
+      mstore(f, _current)
+      mstore(sub(0x20, f), mload(add(BRANCH_DATA_OFFSET, shl(5, 28))))
+      _current := keccak256(0, 0x40)
+
+      f := shl(5, and(_index, shl(29, 1)))
+      mstore(f, _current)
+      mstore(sub(0x20, f), mload(add(BRANCH_DATA_OFFSET, shl(5, 29))))
+      _current := keccak256(0, 0x40)
+
+      f := shl(5, and(_index, shl(30, 1)))
+      mstore(f, _current)
+      mstore(sub(0x20, f), mload(add(BRANCH_DATA_OFFSET, shl(5, 30))))
+      _current := keccak256(0, 0x40)
+
+      f := shl(5, and(_index, shl(31, 1)))
+      mstore(f, _current)
+      mstore(sub(0x20, f), mload(add(BRANCH_DATA_OFFSET, shl(5, 31))))
+      _current := keccak256(0, 0x40)
```

Comment:
This part of the assembly block needs to be modified. For more info check https://github.com/spearbit-audits/connext-nxtp/issues/164#issuecomment-1307507146

Reply:
@Saw-mon-and-Natalie just fixed this - pls take a look when you get a chance!